data/campaigns/The_Rise_Of_Wesnoth/scenarios/04a_The_Swamp_of_Esten.cfg

```diff
@@ -572,10 +634,90 @@ Enter at Your Own Risk!"
             speaker=Sir Ruddry
             message= _ "As far as I know, Sir. We have a large army, and they were pressing all able bodied men and boys into service when I left. That orcish army is huge, but they haven’t met the main body of our forces yet."
         [/message]
+    [/event]
+
+    # If the player already has Sir Ladoc, then Sir Ruddry appears at the start to give the information about the road ahead. But the two knights are really one character, there is dialogue later that's said by whichever of them the player has.
+    # This event fires immediately after Haldric says that he hears a noise, so the screen will have scrolled away from Sir Ruddry.
+    [event]
+        name=exit_sir_ruddry
+
+        [kill]
+            id=Sir Ruddry
+        [/kill]
+
+        [reset_fog]
+            [filter_side]
+                side=1
+            [/filter_side]
+            reset_view=yes
+        [/reset_fog]
+        [redraw]
+            side=1
+            clear_shroud=yes
+        [/redraw]
+
+        [unit]
+            side=3
+            type=Walking Corpse
+            variation=mounted
+            id=Dead Ruddry
+            name= _ "Sir Ruddry"
+            x=35
+            y=35
+        [/unit]
+
+        [event]
+            name=last breath
+            [filter]
+                id=Dead Ruddry
+            [/filter]
+
+            [if]
+                [variable]
+                    name=second_unit.id
+                    equals=Sir Ladoc
+                [/variable]
+
+                [then]
+                    [message]
+                        speaker=Sir Ladoc
+                        # po: "remains" means "dead body", and the speaker has just killed Ruddry's zombie
+                        message= _ "This corpse was the remains of Sir Ruddry, the knight who greeted us earlier. I shall be forever in his debt — had you not gone to the Midlands then I would surely have fallen to the orcish blades, yet you would have slain the necromancers before this noble knight reached the swamp."
+                    [/message]
+                [/then]
+
+                [else]
+                    [message]
+                        speaker=$second_unit.id
+                        # po: "remains" means "dead body", and the speaker has just killed Ruddry's zombie
+                        message= _ "This corpse was the remains of the knight who greeted us earlier."
+                    [/message]
+
+                    [message]
+                        speaker=Sir Ladoc
+                        message= _ "I shall be forever in Sir Ruddry’s debt. Had you not gone to the Midlands then I would surely have fallen to the orcish blades, yet you would have slain the necromancers before this noble knight reached the swamp."
```

Comment:
Will change "Sir Ruddry" to "this noble knight", and the later "this noble knight" to just a pronoun. Unless the player is replaying the campaign, I wouldn't expect them to remember the name that was displayed earlier.